repotracker/repotracker.go

```diff
@@ -289,13 +289,21 @@ func (repoTracker *RepoTracker) StoreRevisions(ctx context.Context, revisions []
 					"project_identifier": ref.Identifier,
 					"revision":           revision,
 				}))
+				if pInfo.Project == nil {
```

Comment:
This makes more sense as an if else for `err != nil`; really we just want to make sure we don't have a nil pointer dereference in the case of `err == nil && project == nil` (which isn't supposed to happen but). No reason to log again here, since it's expected to have a nil project if err != nil.